src/content/drafts/custom-formatter-blog-post.md

```diff
@@ -0,0 +1,351 @@
+    "test": "jest"
+  },
+  // ...other config
+```
+
+Now create a file for the test, `formatter.test.js`. In the test file,
+we're going to use the [ESLint Node.js API](https://eslint.org/docs/latest/developer-guide/nodejs-api).
+Add the following code to `formatter.test.js`:
+
+```js
+const { ESLint } = require('eslint');
+const toml = require('toml');
+
+test("Test TOML formatter", async () => {
+  const eslint = new ESLint();
+  const results = await eslint.lintFiles(['test-data/fullOfProblems.js']);
+
+  const tomlFormatter = await eslint.loadFormatter('./formatter.js')
+  const tomlResults = tomlFormatter.format(results);
+  const tomlBackToJs = toml.parse(tomlResults);
+
+  // Check parsing back to JS
+  expect(tomlBackToJs.errorCount).toBe(3);
+
+  // Uncomment the following line to see the results
+  // console.log(tomlResults);
+});
+```
+
+With everything ready, let's run the test:
+
+```bash
+npm test
+```
+
+In the terminal you should see the following output with the passing test results:
+
+```
+> eslint-formatter-toml@1.0.0 test
+> jest
+
+ PASS  ./formatter.test.js
+  ✓ Test TOML formatter (180 ms)
+
+Test Suites: 1 passed, 1 total
+Tests:       1 passed, 1 total
+Snapshots:   0 total
+Time:        1.04 s, estimated 2 s
+Ran all test suites.
+```
+
+Everything is working as expected!
+
+### 4. Publish to npm
+
+Since we've validated that the custom formatter works the way we expect from the tests,
+we are ready to publish it to npm. Once it's published, anyone on the internet
+who wants to use the custom formatter is able to.
+
+Update the project's `package.json` file with the following information:
+
+| **Field**     | **Value**  |
+|:--------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `"name"`        | A unique string name for the package. The package name **must** begin with `eslint-formatter-`. There cannot be any other packages published on npm with the same name. |
+| `"version"`     | A string version for the package, like `"1.0.0"`. |
+| `"description"` | A string description of the package.|
+| `"main"`        | `"formatter.js"` |
+| `"keywords"`    | `["eslint-formatter", "eslintformatter", "eslint"]`. Add any other keywords you'd like to the array. |
+| `"author"`      | String or object with any author information you'd like to add. For more information, refer to the [npm documention](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#people-fields-author-contributors). |
+
+Your `package.json` should look similar to the following: 
+
+```json
+{
+  "name": "eslint-formatter-toml",
+  "version": "1.0.0",
+  "description": "ESLint custom formatter that publishes the formatter output in TOML",
+  "main": "formatter.js",
+  "keywords": ["eslint-formatter", "eslintformatter", "eslint"],
+  "author": {
+    "name" : "Ben Perlmutter",
+    "email" : "ben@perlmutter.io",
```

Comment:
Just double checking -- are you sure you want your email to appear in this blog post? :)

Reply:
fair nuff. will remove 😄